File: tp-fizz-buzz-java/src/test/java/fr/isima/service/FizzBuzzTest.java
package fr.isima.service;

import org.junit.Before;
import org.junit.Test;

import java.util.AbstractMap.SimpleImmutableEntry;

import static org.assertj.core.api.Assertions.assertThat;

public class FizzBuzzTest {

    private FizzBuzz fizzBuzz;

    @Before
    public void before() {
        this.fizzBuzz = new FizzBuzz();
    }

    @Test
    public void fizzBuzz_with3_returnFizz() {
        assertThat(this.fizzBuzz.fizzBuzz(3)).isEqualTo("Fizz");
    }

    @Test
    public void fizzBuzz_with5_returnBuzz() {
        assertThat(this.fizzBuzz.fizzBuzz(5)).isEqualTo("Buzz");
    }

    @Test
    public void fizzBuzz_with1_return1() {
        assertThat(this.fizzBuzz.fizzBuzz(1)).isEqualTo("1");
    }

    @Test
    public void fizzBuzz_with15_returnFizzBuzz() {
        assertThat(this.fizzBuzz.fizzBuzz(15)).isEqualTo("FizzBuzz");
    }

    @Test
    public void fizzBuzzAll_from1To3_return1And2AndFizz() {
        assertThat(this.fizzBuzz.fizzBuzzAll(1, 3).toMap())
                .hasSize(3)
                .contains(new SimpleImmutableEntry<>(1, "1"),
                        new SimpleImmutableEntry<>(2, "2"),
                        new SimpleImmutableEntry<>(3, "Fizz"));
    }

}

File: tp-fizz-buzz-kotlin/src/test/kotlin/fr/isima/service/FizzBuzzTest.kt
package fr.isima.service

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class FizzBuzzTest {

    @Test
    fun fizzBuzz_with3_returnFizz() {
        assertThat(3.fizzBuzz).isEqualTo("Fizz")
    }

    @Test
    fun fizzBuzz_with5_returnBuzz() {
        assertThat(5.fizzBuzz).isEqualTo("Buzz")
    }

    @Test
    fun fizzBuzz_with1_return1() {
        assertThat(1.fizzBuzz).isEqualTo("1")
    }

    @Test
    fun fizzBuzz_with15_returnFizzBuzz() {
        assertThat(15.fizzBuzz).isEqualTo("FizzBuzz")
    }

    @Test
    fun fizzBuzz_from1To3_return1And2AndFizz() {
        assertThat((1..3).fizzBuzz)
                .isEqualTo(sortedMapOf(1 to "1",
                        2 to "2",
                        3 to "Fizz"))
    }

    @Test
    fun fizzBuzzToPlainText_WithOneItem_ShouldBeEqualToFizzBuzzString() {
        assertThat((10..11).fizzBuzz.plainText).isEqualTo("FizzBuzz all !!!! :\n10 : Buzz\n11 : 11")
    }

}